### PR TITLE
[ENG-6070][cli] generate tvOS credentials for adhoc and prod

### PR DESCRIPTION
**Pull request by quinlanj**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why
- Ensure tvOS projects play nice for adhoc, enterprise and app store builds. 
- Generate the proper credentials for tvOS in the case where the user wants Expo to manage them

# Test Plan

- [x] new tests pass

# Manual sanity check
Since this PR touches many files

- [x] tvOS Adhoc build works
- [x] tvOS App store build works
- [x] iOS App store build works
- [x] iOS Adhoc build works

### CODE REVIEW

**Comment by quinlanj** [comment]
/changelog-entry new-feature add tvOS credentials compatibility for Adhoc and App store builds.

**Review by douglowder — APPROVED** [review]

File: packages/eas-cli/src/credentials/ios/actions/SetUpAdhocProvisioningProfile.ts [hunk 1]
```diff
@@ -109,11 +111,17 @@ export class SetUpAdhocProvisioningProfile {
     );
 
     // 4. Reuse or create the profile on Apple Developer Portal
+    const applePlatform = await getApplePlatformFromSdkRoot(ctx.target);
+    const profileType =
+      applePlatform === ApplePlatform.TV_OS
+        ? ProfileType.TVOS_APP_ADHOC
+        : ProfileType.IOS_APP_ADHOC;
```
Comment (dsokal) [inline packages/eas-cli/src/credentials/ios/actions/SetUpAdhocProvisioningProfile.ts:118]: So this is probably a good example of why `getApplePlatformFromSdkRoot` should return `IOS` by default (and not null).

**Review by dsokal — APPROVED** [review]

File: packages/eas-cli/src/project/ios/__tests__/target-test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      switch (platform) {
+        case ApplePlatform.IOS:
+        case ApplePlatform.TV_OS:
+        case ApplePlatform.MAC_OS:
+          break;
+        default:
+          throw new Error(`Update the function to work with ${platform}`);
+      }
+    }
+  });
```
[the remark this answers is not in the document]
Comment (quinlanj) [reply packages/eas-cli/src/project/ios/__tests__/target-test.ts:28]: yup!

File: packages/eas-cli/src/credentials/context.ts [hunk 3]
```diff
@@ -134,3 +139,12 @@ export class CredentialsContext {
     this.shouldAskAuthenticateAppStore = false;
   }
 }
+
+/**
+ * Credentials context for a specific XCode target.
+ * */
+export class TargetCredentialsContext extends CredentialsContext {
+  constructor(public target: Target, options: CredentialsContextOptions) {
```
[the remark this answers is not in the document]
Comment (quinlanj) [reply packages/eas-cli/src/credentials/context.ts]: yeah that makes sense. passing the extra argument is simpler

**Pull request merged**